client/informant/src/lib.rs

```diff
@@ -19,90 +19,138 @@
 //! Console informant. Prints sync progress and block events. Runs on the calling thread.
 
 use ansi_term::Colour;
-use sc_client_api::{BlockchainEvents, UsageProvider};
 use futures::prelude::*;
-use log::{info, warn, trace};
-use sp_runtime::traits::Header;
-use sc_service::AbstractService;
+use log::{info, trace, warn};
+use parity_util_mem::MallocSizeOf;
+use sc_client_api::{BlockchainEvents, UsageProvider};
+use sc_network::{network_state::NetworkState, NetworkStatus};
+use sp_blockchain::HeaderMetadata;
+use sp_runtime::traits::{Block as BlockT, Header};
+use sp_transaction_pool::TransactionPool;
+use sp_utils::mpsc::TracingUnboundedReceiver;
+use std::fmt::Display;
+use std::sync::Arc;
 use std::time::Duration;
 
 mod display;
 
 /// The format to print telemetry output in.
-#[derive(PartialEq)]
-pub enum OutputFormat {
-	Coloured,
-	Plain,
+#[derive(Clone)]
+pub struct OutputFormat {
+	/// Use colors to display the logs
+	pub colors: bool,
+	/// Add a prefix before every log line
+	pub prefix: String,
```

Comment:
It should here also again should be made aware that this does not uses a whitespace in front of. While I'm still not sure if we should not add the whitespace by default for a prefix.

Another point, please create the following:
```
struct Prefix {
    prefix: String,
    style: Option<ansi_term::Style>,
}
```

This will make it possible for the user to supply a style as well.

Reply:
We should not add the whitespace imo. People might want to use something like `[mynode]` without whitespace in front of the log. But as a counter argument I would also add that the emoji uses a whitespace already so it would be inconsistent. (But I personally find the emojis annoying in logs.)

Reply:
I think the style is not really useful as people can put it in the String already. Do you really want it?